java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitution.java

```diff
@@ -30,6 +30,13 @@
 public class TypeSubstitution {
   private Map<TypeVariableJavaType, JavaType> substitutions = Maps.newLinkedHashMap();
 
+  public TypeSubstitution() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1186)